menas/ui/service/EntityValidationService.js

```diff
@@ -71,4 +71,26 @@ var EntityValidationService = new function () {
     return isOk;
   };
 
+  this.hasValidProperties = function(aProperties) {
+    let isOk = true;
+
+    aProperties.map((oProp) => {
+      //check essentiality - if Mandatory property has no or empty value, we fail
+      if(oProp.essentiality._t === "Mandatory" && !oProp.value) {
```

Comment:
I think it would be worth showing some `validationText` informing about a `Recommended` property having no value, too. I believe this is not addressed.

Reply:
Yes I know... I need to check how to trigger this validation on focus out - currently it happens on the form submission

Reply:
Alright so this is implemented in the last commit, don't hate me for the timeout.

I tried to do this more declaratively, but I'd have to dig deeper into how these binding templates are initialized. I think for now we should try this and if it's causing issues, we'll fix in a separate issue.